File: C#/0905. Sort Array By Parity.cs
public class Solution {
    public int[] SortArrayByParity(int[] A) {
        int left=0;
        int right=A.Length-1;
        while (left<A.Length && A[left]%2==0){
            left++;
        }
        while(right>left &&A[right]%2!=0){
            right--;
        }
        while(left<right){
            if ((A[left])%2==0){
                left++;
            }
            else{
                int tmp=A[left];
                A[left]=A[right];
                A[right]=tmp;
                right--;
            }
        }
        return A;
    }
}

File: 20190519python/0905. Sort Array By Parity.py
class Solution:
    def sortArrayByParity(self, A: List[int]) -> List[int]:
        left,right=0,0
        while right<len(A) and left<len(A):
            while left<len(A) and A[left]%2==0:
                left+=1
            right=left
            while right<len(A) and A[right]%2!=0:
                right+=1
            if left==len(A) or right==len(A):
                break
            A[left],A[right]=A[right],A[left]
            
            right+=1
            left+=1
        return A
